src/util/misc.js

```diff
@@ -681,6 +687,61 @@
       };
     },
 
+    calcRotateMatrix: function(options) {
+      if (!options.angle) {
+        return fabric.iMatrix.concat();
+      }
+      var theta = fabric.util.degreesToRadians(options.angle),
+          cos = fabric.util.cos(theta),
+          sin = fabric.util.sin(theta);
+      return [cos, sin, -sin, cos, 0, 0];
+    },
+
+    calcDimensionsMatrix: function(options) {
+      var scaleX = typeof options.scaleX === 'undefined' ? 1 : options.scaleX,
+          scaleY = typeof options.scaleY === 'undefined' ? 1 : options.scaleY,
+          scaleMatrix = [scaleX, 0, 0, scaleY, 0, 0];
+      if (options.skewX) {
+        scaleMatrix = fabric.util.multiplyTransformMatrices(
+          scaleMatrix,
+          [1, 0, Math.tan(fabric.util.degreesToRadians(options.skewX)), 1],
+          true);
+      }
+      if (options.skewY) {
+        scaleMatrix = fabric.util.multiplyTransformMatrices(
+          scaleMatrix,
+          [1, Math.tan(fabric.util.degreesToRadians(options.skewY)), 0, 1],
+          true);
+      }
+      return scaleMatrix;
+    },
+
+    /**
+     * Returns a transform matrix starting from an object of the same kind of
+     * the one returned from qrDecompose
+     * @static
+     * @memberOf fabric.util
+     * @param  {Object} options
+     * @param  {Number} [options.angle]
+     * @param  {Number} [options.scaleX]
+     * @param  {Number} [options.scaleY]
+     * @param  {Number} [options.skewX]
+     * @param  {Number} [options.skewX]
+     * @param  {Number} [options.translateX]
+     * @param  {Number} [options.translateY]
+     * @return {Array[Number]} transform matrix
+     */
+    componeMatrix: function(options) {
```

Comment:
componentMatrix?

Reply:
i think is 'compose'.
I used it in a precedent implementation of the parsing, i do not need to keep it.
i ll remove it.